### PR TITLE
Fix/update challenges for responsive-web-design-principles

### LINKED ISSUE #35785: Create a media query challenge may fail
Challenge: https://learn.freecodecamp.org/responsive-web-design/responsive-web-design-principles/create-a-media-query/

I have seen issues with this challenge on the forum a number of times.

The [test](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md) is using the computed properties but because you can set a minimum font size in the browser this may not work. In Chrome, you can use `chrome://settings/fonts` and set the minimum font size to be greater than 10px. If you then run the test it will fail.

Changing the test to use a regex instead might be better.

I'm not sure if this proposed change will fix all problem cases but it will at least take care of any minimum font size set in the browser. It should likely also fix any browser compatibility issues. I say that because this is one of the challenges where switching browsers often seem to help people pass the test.

### PR DESCRIPTION
**Pull request by aditijoshi14**
Fixing test for  https://learn.freecodecamp.org/responsive-web-design/responsive-web-design-principles/create-a-media-query/ challenge. 

- [X] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] None of my changes are plagiarized from another source without proper attribution.
- [X] All the files I changed are in the same world language (only English)
- [X] My changes do not use shortened URLs or affiliate links.

Closes #35785

### CODE REVIEW

File: curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md [hunk 1]
```diff
@@ -26,10 +26,13 @@ Add a media query, so that the <code>p</code> tag has a <code>font-size</code> o
 
 ```yml
 tests:
-  - text: Your <code>p</code> element should have the <code>font-size</code> of 10px when the device <code>height</code> is less than or equal to 800px.
-    testString: assert($('p').css('font-size') == '10px', 'Your <code>p</code> element should have the <code>font-size</code> of 10px when the device <code>height</code> is less than or equal to 800px.');
   - text: Declare a <code>@media</code> query for devices with a <code>height</code> less than or equal to 800px.
-    testString: assert(code.match(/@media\s*?\(\s*?max-height\s*?:\s*?800px\s*?\)/g), 'Declare a <code>@media</code> query for devices with a <code>height</code> less than or equal to 800px.');
+    testString: assert($("style").text().replace(/\s/g ,'').match(/@media\(max-height:800px\)/g));
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md]: It is not necessary to escape `{` or `}`.

File: curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md [hunk 2]
```diff
@@ -26,10 +26,13 @@ Add a media query, so that the <code>p</code> tag has a <code>font-size</code> o
...
+  - text: Your <code>p</code> element should have the <code>font-size</code> of 10px when the device <code>height</code> is less than or equal to 800px.
+    testString: assert($("style").text().replace(/\s/g ,'').match(/@media\(max-height:800px\)\{p\{font-size:10px;\}\}/g));
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md]: It is not necessary to escape { or }.  Also, you should make the semi-colon optional if there are no other properties after a property/value pair.

File: curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md [hunk 3]
```diff
@@ -26,10 +26,13 @@ Add a media query, so that the <code>p</code> tag has a <code>font-size</code> o
...
+  - text: Your <code>p</code> element should have initial <code>font-size</code> of 20px when device <code>height</code> is more than 800px.
+    testString: assert($("style").text().replace(/\s/g ,'').replace(/@media.*\}/g, '').match(/p\{font-size:20px;\}/g));
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md]: It is not necessary to escape { or }.

File: curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md [hunk 4]
```diff
@@ -26,10 +26,13 @@ Add a media query, so that the <code>p</code> tag has a <code>font-size</code> o
...
+
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md]: Remove this extra line:
```suggestion
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@aditijoshi14 See my suggested changes.

**Comment by aditijoshi14** [comment]
@RandellDawson  Thank you so much for your suggestions. I made the changes requested.

**Review by RandellDawson — APPROVED** [review]

File: curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md [hunk 5]
```diff
@@ -26,13 +26,13 @@ Add a media query, so that the <code>p</code> tag has a <code>font-size</code> o
 
 ```yml
 tests:
-  - text: Your <code>p</code> element should have the <code>font-size</code> of 10px when the device <code>height</code> is less than or equal to 800px.
-    testString: assert($('p').css('font-size') == '10px', 'Your <code>p</code> element should have the <code>font-size</code> of 10px when the device <code>height</code> is less than or equal to 800px.');
   - text: Declare a <code>@media</code> query for devices with a <code>height</code> less than or equal to 800px.
-    testString: assert(code.match(/@media\s*?\(\s*?max-height\s*?:\s*?800px\s*?\)/g), 'Declare a <code>@media</code> query for devices with a <code>height</code> less than or equal to 800px.');
-
+    testString: assert($("style").text().replace(/\s/g ,'').match(/@media\(max-height:800px\)/g));
+  - text: Your <code>p</code> element should have the <code>font-size</code> of 10px when the device <code>height</code> is less than or equal to 800px.
```
Comment (Manish-Giri) [inline curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md]:
```suggestion
  - text: Your <code>p</code> element should have a <code>font-size</code> of 10px when the device <code>height</code> is less than or equal to 800px.
```

File: curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md [hunk 6]
```diff
@@ -26,13 +26,13 @@ Add a media query, so that the <code>p</code> tag has a <code>font-size</code> o
...
+    testString: assert($("style").text().replace(/\s/g ,'').match(/@media\(max-height:800px\){p{font-size:10px;?}}/g));
+  - text: Your <code>p</code> element should have initial <code>font-size</code> of 20px when device <code>height</code> is more than 800px.
```
Comment (Manish-Giri) [inline curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.english.md]:
```suggestion
  - text: Your <code>p</code> element should have an initial <code>font-size</code> of 20px when the device <code>height</code> is more than 800px.
```

**Review by Manish-Giri — CHANGES_REQUESTED** [review]
Just some minor changes.

**Comment by aditijoshi14** [comment]
@Manish-Giri  Thank you so much for your suggestion. I have committed your requested changed.

**Review by Manish-Giri — APPROVED** [review]
@aditijoshi14 LGTM. Thank You for your contribution. 💯

**Pull request merged**